opensrp-opd/src/main/java/org/smartregister/opd/utils/VisitUtils.java

```diff
@@ -0,0 +1,241 @@
+                    VisitDetail detail = new VisitDetail();
+                    detail.setVisitDetailsId(UUID.randomUUID().toString());
+                    detail.setVisitId(visit.getVisitId());
+                    detail.setVisitKey(obs.getFormSubmissionField());
+                    detail.setParentCode(obs.getParentCode());
+                    detail.setDetails(getDetailsValue(detail, obs.getValues().toString()));
+                    detail.setHumanReadable(getDetailsValue(detail, obs.getHumanReadableValues().toString()));
+                    detail.setCreatedAt(event.getDateCreated() != null ? event.getDateCreated().toDate() : new Date());
+                    detail.setUpdatedAt(event.getDateEdited() != null ? event.getDateEdited().toDate() : detail.getCreatedAt());
+
+                    List<VisitDetail> currentList = details.get(detail.getVisitKey());
+                    if (currentList == null)
+                        currentList = new ArrayList<>();
+
+                    currentList.add(detail);
+                    details.put(detail.getVisitKey(), currentList);
+                }
+            }
+        }
+
+        visit.setVisitDetails(details);
+        return visit;
+    }
+
+    public static void processVisit(EventClient baseEvent) {
+        try {
+            // delete existing events from the database
+            OpdLibrary.getInstance().visitRepository().deleteVisit(baseEvent.getEvent().getFormSubmissionId());
+
+            Visit visit = eventToVisit(baseEvent.getEvent());
+
+            OpdLibrary.getInstance().visitRepository().addVisit(visit);
+            if (visit.getVisitDetails() != null) {
+                for (Map.Entry<String, List<VisitDetail>> entry : visit.getVisitDetails().entrySet()) {
+                    if (entry.getValue() != null) {
+                        for (VisitDetail detail : entry.getValue()) {
+                            OpdLibrary.getInstance().visitDetailsRepository().addVisitDetails(detail);
+                        }
+                    }
+                }
+            }
+            // Update the last interacted with of the user
+            updateLastInteractedWith(baseEvent.getEvent(), visit.getVisitId());
+            abortTransaction();
+
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+    }
+
+    public static void updateLastInteractedWith(@NonNull Event event, @NonNull String visitId) throws
+            CheckInEventProcessException {
+        String tableName = OpdUtils.metadata().getTableName();
+
+        String lastInteractedWithDate = String.valueOf(new Date().getTime());
+
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("last_interacted_with", lastInteractedWithDate);
+
+        int recordsUpdated = OpdLibrary.getInstance().getRepository().getWritableDatabase()
+                .update(tableName, contentValues, "base_entity_id = ?", new String[]{event.getBaseEntityId()});
```

Comment:
Move `currentObject.getEventDate()` and `base_entity_id` to constants.

Reply:
moved `base_entity_id` and `last_interacted_with` to constants